utils/data_reader.py

```diff
@@ -68,6 +101,30 @@ def case_insensitive_lookup(row, key, version):
                 return row[actual_key]
 
 
+def translation_lookup(row, key, version):
+    defaults = {}
+    if type(version) == str:
+        desired_key = DATA_DICTIONARY[version][key]
+    else:
+        desired_key = version.get(key, key)
+        defaults = version.get("default_values", {})
+
+    if desired_key in row:
+        return row[desired_key]
+    else:
+        for actual_key in row.keys():
+            if actual_key.lower() == desired_key.lower():
+                return row[actual_key]
+
+    if desired_key in defaults:
+        return defaults[desired_key]
+    else:
+        for actual_key in defaults.keys():
+            if actual_key.lower() == desired_key.lower():
+                return defaults[actual_key]
```

Comment:
This code block is duplicated [3 times](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) - twice here and once in `case_insensitive_lookup`.

A simple helper method that takes in `desired_key` and `row`

Reply:
I'm not sure we can have our cake and eat it too in this case. The code block that is repeated is `case_insensitive_lookup` itself, which was extricated from what is now `tranlsation_lookup` so as to reduce complexity. The only reason I can't call `case_insensitive_lookup` from `translation_lookup` is that `case_insensitive_lookup` uses a dictionary lookup to find the 'desired key' from the global `DATA_DICTIONARY` which I don't want to do if translating a `.csv`. I could rename what is now `case_insensitive_lookup` to something like `schema_lookup` make it two lines, the `DATA_DICTIONARY` lookup and then a call to the spun off part of `case_insensitive_lookup` that actually does a case insensitive lookup, which could also then be referenced in `translation_lookup` but then `schema lookup` seems trivial and should be folded into `translation_lookup`, of which it is a special case. This, I'm sure it goes without saying, gets us right back to where we started...unless you see things differently.

Reply:
Addressed in https://github.com/mitre/data-owner-tools/pull/24/commits/d4c4da99851a442ed1d55b1e90e641b198c1540f